bodo/numba_compat.py

```diff
@@ -5739,6 +5693,11 @@ def define_untyped_pipeline(state, name="untyped"):
     if state.flags.enable_ssa:
         pm.add_pass(ReconstructSSA, "ssa")
 
+    # Bodo change: disable extra branch pruning since it can result in IR issues
+    # See test_batched_read_agg for example (TODO(Ehsan): investigate and re-enable)
+    # if not state.flags.no_rewrites:
+    #     pm.add_pass(DeadBranchPrune, "dead branch pruning")
```

Comment:
Remove?

Reply:
Keeping it here in addition to the comment to make it clear what we changed from Numba and why. This context is important in the next upgrade that touches this.